zenoh/src/net/routing/mod.rs

```diff
@@ -168,4 +169,18 @@ impl RoutingContext<NetworkMessage> {
         }
         None
     }
+
+    #[inline]
+    pub(crate) fn full_key_expr(&self) -> Option<OwnedKeyExpr> {
+        match self.full_expr() {
+            Some(full_expr) => {
+                if let Ok(keyexpr) = OwnedKeyExpr::new(full_expr) {
+                    Some(keyexpr)
+                } else {
+                    None
+                }
+            }
+            None => None,
+        }
```

Comment:
To improve readability you may want to look into [?](https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html) and [ok](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok).

Code would then be reduced to:
```rust
let full_expr = self.full_expr()?;
OwnedKeyExpr::new(full_expr).ok()
```